### PR TITLE
Constant propagation of integer to string primitives

### PR DESCRIPTION
**Pull request by danpoe**
Implement constant propagation of int to string and long to string primitives, i.e. function application expressions with IDs ID_cprover_string_of_int_func and ID_cprover_string_of_long_func.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-symex/goto_symex.cpp [hunk 1]
```diff
@@ -503,3 +508,77 @@ bool goto_symext::constant_propagate_string_substring(
 
   return true;
 }
+
+bool goto_symext::constant_propagate_integer_to_string(
```
Comment (smowton) [inline src/goto-symex/goto_symex.cpp:513]: Document the expected arguments at the top of this function -- looks like `string_length, string_data, value, [base]`?

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: acc737a4825f2985d9b721b12b1a39639e3b161d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/125438420

File: src/goto-symex/goto_symex.cpp (hunk 1 above)
Comment (danpoe) [reply src/goto-symex/goto_symex.cpp:513]: Done

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 90e7266186256e9c3677b92fa0e148f8adc96bb4).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/125447439

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 839c86e3e8d3d438497dfe29fb3f0a46688ca0f3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/125447258
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Pull request merged**